uiowa/uiowa
# P2LB protected node revision updates

Resolves #6859 

# How to test

```
ddev blt sync --site hawkeyemarchingband.uiowa.edu && ddev drush @hawkeyemarchingband.local uli admin/content?type=page
```
- Create a draft version of one or two pages without publishing so that the draft is the newest version of the page. It is easier to test this if your draft includes an edit that makes it immediately visually clear so you can see the change when you visit the converted page.
- Activate the split
```
ddev drush @hawkeyemarchingband.local config-split:activate p2lb -y 
```
- Visit https://hawkeyemarchingband.uiowa.ddev.site/admin/content/sitenow-converter
- Convert the pages which you modified.
- Confirm that the converted pages have the changes you made to the drafts.
- View the revisions and confirm that the draft(s) exist still.

## Cleanup

- Visit https://hawkeyemarchingband.uiowa.ddev.site/admin/content/sitenow-converter again.
- Convert the remaining pages in bulk.
- Visit https://hawkeyemarchingband.uiowa.ddev.site/admin/content/sitenow-converter/converted?moderation_state=editorial-draft.
- Bulk publish all pages.
```
ddev drush @hawkeyemarchingband.local sitenow_p2lb:cleanup
```
- Ensure the process completes.

# Bonus
- Test the whole process on Brand!
```
ddev blt sync --site brand.uiowa.edu && ddev drush @brand.local config-split:activate p2lb -y && ddev drush @brand.local uli admin/content/sitenow-converter
```
- https://brand.uiowa.ddev.site/admin/content/sitenow-converter/converted?moderation_state=editorial-draft
```
ddev drush @brand.local sitenow_p2lb:cleanup
```
